### PR TITLE
take and hold a reference to ChannelManager

### PR DESCRIPTION
**Pull request by johncantrell97**
There's no real need for the LiquidityManager to actually hold the channel manager.  Once LSPS1 and/or LSPS2 are landed we should remove the reference from LiquidityManager as the reference will just pass through to the constructors of the protocols.

Instead of putting the type constraints on just the `new` method, this way is closer to what it will actually look like once the `LiquidityManager` will hold LSPS1/2 objects that have these same generics.

### CODE REVIEW

**Comment by tnull** [comment]
The CI failure should be due to https://github.com/lightningdevkit/ldk-lsp-client/pull/10. Feel free to rebase this PR after it has been merged.

**Comment by johncantrell97** [comment]
Rebased, thanks

File: src/transport/message_handler.rs [hunk 1]
```diff
@@ -41,25 +45,53 @@ pub struct LiquidityProviderConfig;
 ///
 /// Should be used as a [`CustomMessageHandler`] for your
 /// [`lightning::ln::peer_handler::PeerManager`]'s [`lightning::ln::peer_handler::MessageHandler`].
-pub struct LiquidityManager<ES: Deref>
-where
+pub struct LiquidityManager<
+	ES: Deref,
```
Comment (tnull) [inline src/transport/message_handler.rs:49]: @johncantrell97 Did you check that dropping all those `Send + Sync` bounds is indeed fine?

Comment (johncantrell97) [reply src/transport/message_handler.rs:49]: nope, need to rebase lsps2 on top of this so I can check but wanted to make sure this idea was going to be accepted before rebasing

Comment (johncantrell97) [reply src/transport/message_handler.rs:49]: ok just went through everything with c= node and we don't need any of the Send + Sync + 'static bounds at all.

Comment (tnull) [reply src/transport/message_handler.rs:49]: Great!

**Review by tnull — APPROVED** [review]

**Pull request merged**